### PR TITLE
Introduce toTable() API

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
The `readableCsvChannel.toTable()` method returns a table from a readableCSVChannel. This method performs the same functionality as `readableCsvChannel.getTable()` (which will be deprecated), but makes the `keyFieldNames` parameter mandatory. 

Related issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/1871

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

**Comment by BuddhiWathsala** [comment]
@daneshk @kaneeldias are we going to introduce a new API for this?. I thought we are going to revamp these old APIs.

**Comment by daneshk** [comment]
> @daneshk @kaneeldias are we going to introduce a new API for this?. I thought we are going to revamp these old APIs.

Revamp the old API is a breaking change. Ideally what we want is to mandate passing field keys and the parameter name is not misleading. So we thought of introducing a new one and deprecating the old APIs. WDYT?

**Comment by BuddhiWathsala** [comment]
Yes, that would work then. +1.

File: ballerina/readable_csv_channel.bal [hunk 1]
```diff
@@ -120,16 +120,29 @@ public class ReadableCSVChannel {
     # ```
     #
     # + structType - The object in which the CSV records should be deserialized
-    # + fieldNames - The names of the fields used as the (composite)key of the table
+    # + fieldNames - The names of the fields used as the (composite) key of the table
     # + return - Table, which represents the CSV records or else an `io:Error`
```
Comment (BuddhiWathsala) [inline ballerina/readable_csv_channel.bal:124]: Don't we need `deprecated` annotation here?

File: ballerina/tests/csv_io.bal [hunk 2]
```diff
@@ -525,6 +532,33 @@ isolated function testTableContent() {
     }
 }
 
+@test:Config {}
+isolated function testTableContent2() {
+    string filePath = RESOURCES_BASE_PATH + "datafiles/io/records/sample5.csv";
+    float expectedValue = 60001.00;
+    float total = 0.0;
+
+    var csvChannel = openReadableCsvFile(filePath);
```
Comment (ayeshLK) [inline ballerina/tests/csv_io.bal:541]: Shall we update the method signature with `returns error?` and use `check` here ? Then we could remove unwanted if-else conditions. Check for other places as well.

File: ballerina/readable_csv_channel.bal (hunk 1 above)
Comment (kaneeldias) [reply ballerina/readable_csv_channel.bal:124]: Fixed in 4b05dc9

File: ballerina/tests/csv_io.bal (hunk 2 above)
Comment (kaneeldias) [reply ballerina/tests/csv_io.bal]: Fixed in 54974f8

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**